### PR TITLE
[ Frontend ] Refactored frontend code🔨  / State Management -Redux 🚀

### PR DESCRIPTION
**Pull request by Vishv07**
This PR includes 3 new Dependency for redux. Kindly check on package.json
 

- Added Redux for better state management.
- Added services for Axios and sockets.
- Reduced Props drilling.
- Count for Participants 
- Centralized Snackbar. 

As we have now proper state management so we can build new features on top of them easily.

I request @kothariji  and other people to review this PR thoroughly. Let me know if you have any doubts and Changes in your mind 👍. 

I'll post some guidelines on our discord channel and here too for following some coding guidelines for good coding standards.

### CODE REVIEW

**Review by karan0805 — APPROVED** [review]

File: frontend/syntaxmeets/src/components/SyntaxChat/SyntaxChat.js [hunk 1]
```diff
@@ -33,37 +34,35 @@ const SyntaxChat = (props) => {
   const [typingUser, setTypingUser] = useState();
   const [emojiPickerState, SetEmojiPicker] = useState(false);
   const messagesEndRef = useRef(null);
-
-  const [mCount, setMCount] = useState(0);
-  const [state, setState] = useState(false);
+  const [openDrawer, setopenDrawer] = useState(false);
 
   const handleMessageSubmit = () => {
-    SetEmojiPicker(false);
```
Comment (kothariji) [inline frontend/syntaxmeets/src/components/SyntaxChat/SyntaxChat.js:41]: Add this function also

**Review by kothariji — APPROVED** [review]
સિંટેક્સમિટ્સ ની આખી ટીમ તરફથી @Vishv07 , હાર્દિક આભાર❤️
અમારા પ્રોજેક્ટ માટે તમારી સખત મહેનત અને સમર્પણની અમે ખરેખર પ્રશંસા કરીએ છીએ. અમે આશા રાખીએ કે તમે ઘણી સફળતા પ્રાપ્ત કરશો. 🙌

*Translate: English 😊 *
@Vishv07, a Heartfelt Gratitude, from the entire team of SyntaxMeets ❤️ 
We really appreciate your hard work and dedication towards our project. We hope you achieve a lot of success. 🙌

**Comment by Vishv07** [comment]
@kothariji This by far the best comment I have ever received.🥺.  ધન્યવાદ 🙏. ખુબ ખુબ આભાર તમારો. 
Looking forward to working on cool things with you and your team.🚀

**Comment by kothariji** [comment]
> @kothariji This by far the best comment I have ever received.🥺. ધન્યવાદ 🙏. ખુબ ખુબ આભાર તમારો.
> Looking forward to working on cool things with you and your team.🚀

Yesss 🙌🙌

**Pull request merged**